src/test/java/com/google/gcp_variant_transforms/library/BigQueryRowGeneratorTest.java

```diff
@@ -0,0 +1,140 @@
+
+  @Inject
+  public VcfParser vcfParser;
+
+  @Inject
+  public BigQueryRowGenerator bigQueryRowGenerator;
+
+  @Before
+  public void buildBigQueryRowsFromVCFFile() throws FileNotFoundException {
+    File smallFile = new File(TEST_FILE);
+    List<String> fileLines = IOUtil.slurpLines(smallFile);
+    final int firstNonComment = IntStream.range(0, fileLines.size()).
+            filter(i -> !fileLines.get(i).startsWith("#")).findFirst().getAsInt();
+    ImmutableList<String> headerLines = ImmutableList.copyOf(fileLines.subList(0,
+            firstNonComment));
+    vcfCodec = vcfParser.generateCodecFromHeaderLines(headerLines);
+    vcfHeader = vcfCodec.getHeader();
+    // all records in VCF file
+    vcfRecords = ImmutableList.copyOf(
+            fileLines.subList(firstNonComment, fileLines.size()));
+
+    // build BQ rows
+    rows = new ArrayList<>();
+    for (int i = 0; i < vcfRecords.size(); ++i) {
+      rows.add(bigQueryRowGenerator.getRows(vcfCodec.decode(vcfRecords.get(i)), vcfHeader));
+    }
+  }
+
+  /**
+   * Records in the test VCF file:
+   * #CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	NA00001	NA00002	NA00003
+   * 20	14370	rs6054257	G	A	29	PASS	NS=3;DP=14;AF=0.5;DB;H2	GT:GQ:DP:HQ	0|0:48:1:51,51
+   * 1|0:48:8:51,51	1/1:43:5:.,.
+   * 20	17330	.	T	A	3	q10	NS=3;DP=11;AF=0.017	GT:GQ:DP:HQ	0|0:49:3:58,50	0|1:3:5:65,3	0/0:41:3
+   * 20	1110696	rs6040355	A	G,T	67	PASS	NS=2;DP=10;AF=0.333,0.667;AA=T;DB	GT:GQ:DP:HQ
+   * 1|2:21:6:23,27	2|1:2:0:18,2	2/2:35:4
+   * 20	1230237	.	T	.	47	PASS	NS=3;DP=13;AA=T	GT:GQ:DP:HQ	0|0:54:7:56,60	0|0:48:4:51,51	0/0:61:2
+   * 19	1234567	microsat1	GTCT	G,GTACT	50	PASS	NS=3;DP=9;AA=G	GT:GQ:DP	0/1:35:4	0/2:17:2
+   * 1/1:40:3
+   * There are 5 records in the file, thus there will be 5 BQ rows
+   * The test should cover all kinds of elements in those records, including corner cases.
+   */
+  @Test
+  public void testBigQueryRowGenerator_whenCheckingRowElements_thenTrue() {
+    assertThat(rows.size()).isEqualTo(5);
+    // The first row covers all base fields without missing field value("."),
+    // and this row also contains record HQ=".,.", which should be [null,null] in the BQ row
+    TableRow firstRow = rows.get(0);
+    assertThat(firstRow.get(VariantToBqUtils.ColumnKeyConstants.REFERENCE_NAME))
+            .isEqualTo("20");
+    assertThat(firstRow.get(VariantToBqUtils.ColumnKeyConstants.START_POSITION))
+            .isEqualTo(14370);
+    assertThat(firstRow.get(VariantToBqUtils.ColumnKeyConstants.END_POSITION))
+            .isEqualTo(14370);
+    assertThat(firstRow.get(VariantToBqUtils.ColumnKeyConstants.NAMES))
+            .isEqualTo("rs6054257");
+    assertThat(firstRow.get(VariantToBqUtils.ColumnKeyConstants.REFERENCE_BASES)).isEqualTo("G");
+
+    // AF field has type=`A` in VCFHeader, which should be moved from Info field to ALT field
+    List<TableRow> altMetadata = (List<TableRow>) firstRow.get(
```

Comment:
This test is very large. Is there a way to break up the asserts into multiple functions?
ex. function ideas: 
- test for first row constant fields
- test for first row filter and NS/DP/DB/H2
- test for calls
- test the record with empty fields

Reply:
Separated